src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,168 @@
+                }
+                return new MarkCommand(Integer.parseInt(ops[1]), true);
+            case UNMARK:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_UNMARK_CMD);
+                }
+                return new MarkCommand(Integer.parseInt(ops[1]), false);
+            case DELETE:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_DELETE_CMD);
+                }
+                return new DeleteCommand(Integer.parseInt(ops[1]));
+            case DATE:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_DATE_CMD);
+                }
+                return new ListCommand(DateUtil.toLocalDateTime(ops[1]));
+            case FIND:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_DATE_CMD);
+                }
+                return new ListCommand(ops[1]);
+            default:
+                break;
+            }
+
+            // handle commands with multiple args
+            String[] args;
+            switch (command) {
+            case TODO:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_TODO_CMD);
+                }
+
+                return new AddCommand(DukeCommand.TODO, ops[1], false);
+            case DEADLINE:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_DATE_CMD);
+                }
+
+                args = ops[1].split(" /[a-z]*[^ ] ");
+                if (args.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_DEADLINE_CMD);
+                }
+
+                return new AddCommand(DukeCommand.DEADLINE, args[0], false, DateUtil.toLocalDateTime(args[1]));
+            case EVENT:
+                if (ops.length != 2) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_EVENT_CMD);
+                }
+
+                args = ops[1].split(" /[a-z]*[^ ] ");
+                if (args.length != 3) {
+                    throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_EVENT_CMD);
+                }
+
+                return new AddCommand(DukeCommand.EVENT, args[0], false, DateUtil.toLocalDateTime(args[1]),
+                        DateUtil.toLocalDateTime(args[2]));
+            default:
+                throw new NoSuchCommandException(Message.EXCEPTION_NOSUCH_COMMAND);
+            }
+        } catch (DateTimeParseException e) {
+            throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_CMD_ARGS);
+        } catch (NumberFormatException e) {
+            throw new InvalidCommandArgsException(Message.EXCEPTION_INVALID_TASK_ID_FORMAT);
+        }
+
```

Comment:
I like how you separate the code blocks based on the logic of arguments. However, since the method is rather long, you might want to consider separating the logic each of code block out into another methods?